### DISCUSSION TITLE
Label custom drawing

### DISCUSSION THREAD

**Discussion by ionuteni** [Mapsui/Mapsui#1574] [Q&A] [answered]
Hello, 

I need to place labels(along line-string - more like a curved text following the polyline or inside polygon).
I also need to display them only when they actually fit inside the polygon or along the line, at given viewport/resolution.
Is there a way to override the default label drawing on viewport changed?

Thanks,
Ionut

**Comment by pauldendulk** [comment, marked as answer]
The CustomStyleRenderer is intended for this.

https://mapsui.com/documentation/custom-style-renders.html

**Reply by ionuteni** [reply]
Thanks for pointing me in the right direction, somehow I missed that part from the documentation.
